src/protocol/data/memcache/response.h

```diff
@@ -80,6 +82,10 @@ struct response {
 
     struct bstring          key;        /* key string */
     struct bstring          vstr;       /* value string */
+    char                    *vbuf;      /* vbuf is a buffer of RSP_VAL_BUF_SIZE that processors can use by
```

Comment:
I think in a stable version this needs to be a server decision. So the capacity of the buffer will be announced to the storage module and storage can return the size of the data while copying things into this buffer. The treatment of return value in `snprintf` can be borrowed- a greater than capacity value means copy is incomplete. 

`vstr` above can be reused perhaps? `vstr.len` can carry the capacity of the buffer when passed in, `vstr.data` points to the buffer; and upon return, `vstr.len` can be updated to mean the number of bytes intended. The caller should have a copy of the original value of the buffer size so it can compare the return with that copy to see if further read is necessary to complete the response.